.github/workflows/build.yml

```diff
@@ -1,22 +1,42 @@
 # -----------------------------------------------------------------------------
 #   - invoked on push, pull_request, or manual trigger
-#   - test under at least 4 versions of python
+#   - test under all supported versions of Python
 # -----------------------------------------------------------------------------
 name: build
 on: [push, pull_request, workflow_dispatch]
 
 jobs:
+  get-python-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11  # Just used for running the versions script
```

Comment:
This would need to be maintained manually then? Is there a way to automate this as well?

Reply:
Yes, though we were already doing this anyway. I will look into somehow automating this. The problem of course is that you need some version of python installed in order to run the python version getter script.

But that's not really doing anything fancy so my hope was that 3.11 would work for a very long time